go/exocom-mock/exocom-mock.go

```diff
@@ -61,31 +64,6 @@ func (e *ExoComMock) HasConnection() bool {
 	return e.socket != nil
 }
 
-// CloseConnection closes any open connection
-func (e *ExoComMock) CloseConnection() error {
-	if e.HasConnection() {
-		err := e.socket.Close()
-		if err != nil {
-			return err
-		}
-		e.socket = nil
-	}
-	return nil
-}
-
-// Reset closes and nils the socket and clears all received messages
-func (e *ExoComMock) Reset() error {
-	if e.HasConnection() {
-		err := e.socket.Close()
-		e.socket = nil
-		if err != nil {
-			return err
-		}
-	}
-	e.ReceivedMessages = []structs.Message{}
-	return nil
-}
```

Comment:
These methods were no longer needed as we are no longer resetting the mock anywhere. We are using a new mock in every test and in the retry test we are now closing the mock and opening another one